pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -72,6 +67,12 @@ in buildPythonPackage rec {
   PYTORCH_BUILD_VERSION = version;
   PYTORCH_BUILD_NUMBER = 0;
 
+  USE_FBGEMM = 0; # this can't build because of CMAKE downloads
+  USE_MKLDNN = 0; # mkl with cuda is broken
```

Comment:
MKL with cuda is _not_ broken, actually -- just MKLDNN. The problem seems to be a warning which is treated as an error:
```
/build/source/third_party/ideep/mkl-dnn/src/cpu/jit_uni_reorder_utils.cpp: In function ‘void mkldnn::impl::cpu::tr::prb_normalize(mkldnn::impl::cpu::tr::prb_t&)’:
/build/source/third_party/ideep/mkl-dnn/src/cpu/jit_uni_reorder_utils.cpp:273:29: error: array subscript is above array bounds [-Werror=array-bounds]
                 || p.nodes[j].os < p.nodes[min_pos].os
                    ~~~~~~~~~^
/build/source/third_party/ideep/mkl-dnn/src/cpu/jit_uni_reorder_utils.cpp:276:37: error: array subscript is above array bounds [-Werror=array-bounds]
                         && p.nodes[j].n < p.nodes[min_pos].n);
                            ~~~~~~~~~^
cc1plus: all warnings being treated as errors
make[2]: *** [third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/build.make:1272: third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/jit_uni_reorder_utils.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

Reply:
Found the issue: https://github.com/pytorch/pytorch/issues/22346

Reply:
I missed the workaround at https://github.com/NixOS/nixpkgs/blob/fb76f093c9ab204fbec9dc9c56cb9b4a81f43bc7/pkgs/development/python-modules/pytorch/default.nix#L79

I'll be adding a `mklSupport ? false` which will swap out the numpy BLAS implementation so that this is more explicit. The working code is similar to `my_magma`:

```
  my_numpy = if mklSupport && numpy.blasImplementation != "mkl" then numpy.override { blas = mkl; } else numpy;
```